src/Enum.php

```diff
@@ -182,9 +190,9 @@ protected static function labels()
     }
 
     /**
-     * @param string|int $input
+     * @param TEnumValue|string $input
      *
-     * @return \Spatie\Enum\EnumDefinition|null
+     * @return \Spatie\Enum\EnumDefinition<TEnumValue>|null
```

Comment:
I really don't know but shouldn't this be `EnumDefinition<string, TEnumValue, string>`? 🤔
Or how does this work?

Reply:
I see what you're saying, and not quite.

From the docs:
> @template tag order matters for class docblocks, as they dictate the order in which those generic parameters are referenced in docblocks

As such, the bit between the angle-brackets is based on the `@template` annotations as opposed to the content of the constructor.

In other languages, this is equivalent to
```java
class Foo<T>{
  public foo(String a, T b, String c);
}```

Where the instantiated type would be `Foo<String>`

Reply:
Okay. Thanks for explaining!
I think that I will merge & release it on Monday.

Have a nice weekend! 😊